src/server.ts

```diff
@@ -69,18 +69,26 @@ async function callWhenConnected (this: any, method, payload) {
     })
   }
 
-  // Do snazzy logging
-  return promise.then(responseRaw => {
-    const response = (responseRaw && typeof responseRaw === 'string') ? JSON.parse(responseRaw) : responseRaw
-    console.log("")
-    console.log(`WS call: ${method}`.dim.inverse)
-    console.log('request'.green.bold, `------>`.green.bold, `(${typeof payload})`.green.italic)
-    console.log(JSON.stringify(payload, null, 2))
-    console.log('response'.cyan.bold, `<-----`.cyan.bold, `(${typeof response})`.cyan.italic)
-    console.log(JSON.stringify(response, null, 2))
-    console.log("")
+  try {
+    responseRaw = await promise
+  } catch (e) {
+    failure = e
+  }
+
+  const response = (responseRaw && typeof responseRaw === 'string') ? JSON.parse(responseRaw) : responseRaw
+  console.log("")
+  console.log(failure ? `WS call (ERROR): ${method}`.red.inverse : `WS call: ${method}`.dim.inverse)
+  console.log('request'.green.bold, `------>`.green.bold, `(${typeof payload})`.green.italic)
+  console.log(JSON.stringify(payload, null, 2))
+  console.log('response'.cyan.bold, `<-----`.cyan.bold, `(${typeof response})`.cyan.italic)
+  console.log(JSON.stringify(response, null, 2))
+  console.log("")
+
+  if (failure) {
+    throw failure
+  } else {
```

Comment:
Unnecessary else with throw

Reply:
Again, in the absence of a style guide, it comes down to preference. Happy if you want to suggest a style guide / linter.

Reply:
Yeah, this is something I'd love to have in place.
As programmers we shouldn't think about the code style, and should offload this thinking to the tools. :)
Think we'll have this discussion in hAppy team at some point - would love to include you in it too! :)